pkg/server/openapiv3/servicecache.go

```diff
@@ -101,15 +103,20 @@ func (c *ServiceCache) RegisterStaticAPIs(cont *restful.Container) error {
 		byGVPath[gvPath] = []*restful.WebService{t}
 	}
 	for gvPath, ws := range byGVPath {
-		spec, err := builder3.BuildOpenAPISpecFromRoutes(restfuladapter.AdaptWebServices(ws), c.config)
-		if err != nil {
-			return fmt.Errorf("failed to build OpenAPI v3 spec for %s: %w", gvPath, err)
-		}
-		etag, err := computeEtag(spec)
-		if err != nil {
-			return fmt.Errorf("failed to compute OpenAPI v3 spec etag for %s: %w", gvPath, err)
+		c.staticSpecs[gvPath] = &openapi.StaticValue[*spec3.OpenAPI]{
+			GetFunc: func() (value *spec3.OpenAPI, etag string, err error) {
+				spec, err := builder3.BuildOpenAPISpecFromRoutes(restfuladapter.AdaptWebServices(ws), c.config)
+				if err != nil {
+					return nil, "", fmt.Errorf("failed to build OpenAPI v3 spec for %s: %w", gvPath, err)
+				}
+				etag, err = computeEtag(spec)
+				if err != nil {
+					return nil, "", fmt.Errorf("failed to compute OpenAPI v3 spec etag for %s: %w", gvPath, err)
+				}
+
+				return spec, etag, nil
+			},
```

Comment:
this doesn't look correct. We should only call `GetFunc` once, i.e. probably this will fit here too:
```go
cached.Once(cached.Func[*spec3.OpenAPI](
			func() (value *spec3.OpenAPI, etag string, err error) {
...
```